aws/get_running_instances.py

```diff
@@ -0,0 +1,88 @@
+"""
+Use boto to get the number of running or pending EC2 instances matching a key
+name.
+
+Usage:
+    If you've defined the AWS credentials as environment variables or in a
+    .boto file, then use:
+        `python get_running_instances.py -k some-key-name`
+
+    Else, you can add the aws keys as arguments to the above command:
+        ** '--aws-access-key-id' or '-i'
+        ** '--aws-secret-access-key' or '-s'
+"""
+import argparse
+import boto
+import logging
+import os
+import sys
+
+
+logging.basicConfig(format='[%(levelname)s] %(message)s')
+logger = logging.getLogger(__name__)
+
+
+def get_running_instances(key_name, aws_access_key_id, aws_secret_access_key):
+    """
+    Logs and returns (as a string) the number of EC2 instances matching
+    `key_name` that are running or pending.
+
+    :Args:
+        key_name: the ec2 key_name to match.
+        aws_access_key_id
+        aws_secret_access_key
+    """
+    ec2conn = boto.connect_ec2(aws_access_key_id, aws_secret_access_key)
+    reservations = ec2conn.get_all_instances()
+    instances = [
+        i for r in reservations for i in r.instances
+        if i.state in ('running', 'pending') and i.key_name == key_name
```

Comment:
It could be useful, but I don't think it's as important. I mainly wanted to ask because I wasn't sure what Jenkins pays attention to. If shutting-down is ignored by Jenkins, then we should ignore it in this script, too. So...let's not-include it!